### PR TITLE
Fragment support

### PR DESCRIPTION
**Pull request by nicklockwood**

### CODE REVIEW

**Comment by tonyarnold** [comment]
tl;dr LGTM :shipit: 

Perfect! Exactly what I'd hoped for. There are a few situations where this doesn't work, but I'm not sure that it should, under those circumstances, i.e.:

```swift
class PropertiesPanel: NSWindow {

    override var acceptsFirstResponder: Bool {
        return true
    }
}
```

If I select the function body and the preceding line in the class above, and format as a fragment it will drop back to an origin of column 0 (because the preceding line starts at column 0). I guess this is one of the times using an AST-based parser might be benefit (although that would need to parse the entirety of the document…)

Maybe I should be doing a double pass in the source formatter extension — once to get the correct indent of the target selection, then again to just format the selection at the correct indent. Formatting is hard. 

I think for now, what you've done here is great.

**Comment by nicklockwood** [comment]
I think that's easily solved by just ignoring any leading blank lines in the fragment - I probably should've thought of that TBH ¯\_(ツ)_/¯

**Comment by nicklockwood** [comment]
@tonyarnold try it now :-)

**Comment by tonyarnold** [comment]
:kisses_fingers: perfect! Thanks buddy!
